### PR TITLE
Divisions

### PR DESCRIPTION
**Pull request by danil-topc**
#11

### CODE REVIEW

**Comment by danil-topc** [comment]
Now divisions page looks like this with 3 lists: for Normal divisions (exists in ldap and in config), Config only divisions, ldap only divisions. With ability to create Config only divisions and delete ldap only divisions
Screenshot from 2019-05-06 02-36-28

**Comment by matejak** [comment]
This looks great, please add the related bits into documentation as well.

File: src/web/backend/ldap/api.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+class DivisionsListViewSet(EdapMixin, MethodView):
+
+    def get(self):
+        """ Get divisions """
+        ldap_divisions = self.edap.get_divisions()
+        config_divisions = get_config_divisions()
+
+        divisions = {}
+
+        for machine_name, display_name in config_divisions.items():
```
Comment (matejak) [inline src/web/backend/ldap/api.py]: I just have a minor remark - please make a function/method `merge_divisions` that would perform the merge into the dictionary.
That way, the merge functionality can be unit-tested.

File: src/web/backend/nextcloud/utils.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    otherwise just give access to group_name, create subfolder in folder for group_type with group_name
+    Args:
+        nxc (NextCloud): nextcloud instance
+        group_name (str): group name
+        group_type (str): group type
+
+    Returns (bool):
+    """
+    # check if folder for type is already created
+    group_folders = nxc.get_group_folders().data
```
Comment (matejak) [inline src/web/backend/nextcloud/utils.py]: Please make this a separate method and remove the comment (e.g. `get_group_folder(group_type)`).

File: src/web/backend/ldap/api.py (hunk 1 above)
Comment (danil-topc) [reply src/web/backend/ldap/api.py]: You right, done

File: src/web/backend/nextcloud/utils.py (hunk 2 above)
Comment (danil-topc) [reply src/web/backend/nextcloud/utils.py]: done

**Comment by danil-topc** [comment]
@matejak thanks for review, made requested changes. About description to documentation, you mean to web folder Readme ?

**Comment by matejak** [comment]
Yes, I am not aware of any other documentation.
BTW could you write some small unit-tests for the `merge_divisions` function? You can do it in a separate PR though.

**Comment by danil-topc** [comment]
@matejak added description to documentation and unit-test for `merge_divisions` as well.

**Comment by matejak** [comment]
Looks great, thanks!

**Pull request merged**